src/Calendar/CalendarProvider.php

```diff
@@ -0,0 +1,651 @@
+            );
+        } catch (Exception $err) {
+            if ($err instanceof \GuzzleHttp\Exception\ClientException) {
+                $response = $err->getResponse();
+                $statusCode = $response->getStatusCode();
+
+                switch ($statusCode) {
+                    case 403:
+                        throw $this->errorFactory->create(CalendarErr::UPDATE_CALENDAR_EVENT_FORBIDDEN);
+                    case 404:
+                        throw $this->errorFactory->create(CalendarErr::CALENDAR_OR_CALENDAR_EVENT_NOT_FOUND);
+                }
+            }
+            throw ErrorHelper::getSherlError(
+                $err,
+                $this->errorFactory->create(CalendarErr::UPDATE_CALENDAR_EVENT_FAILED)
+            );
+        }
+    }
+
+    /**
+     * Deletes a calendar event request.
+     *
+     * @param string $calendarId The ID of the calendar.
+     * @param string $calendarEventId The ID of the calendar event.
+     * @throws SherlException If there is an error on deleting a calendar event.
+     * @return bool Returns true on success.
+     */
+    public function deleteCalendarEventRequest(string $calendarId, string $calendarEventId): bool
+    {
+        try {
+            $response = $this->client->delete('/api/calendar/' . $calendarId . '/event/' . $calendarEventId, [
+              "headers" => [
+                "Content-Type" => "application/json",
+              ],
+            ]);
+            return filter_var($response->getBody()->getContents(), FILTER_VALIDATE_BOOLEAN);
+        } catch (Exception $err) {
+            if ($err instanceof \GuzzleHttp\Exception\ClientException) {
+                $response = $err->getResponse();
+                $statusCode = $response->getStatusCode();
+
+                switch ($statusCode) {
+                    case 403:
+                        throw $this->errorFactory->create(CalendarErr::DELETE_CALENDAR_EVENT_FORBIDDEN);
+                    case 404:
+                        throw $this->errorFactory->create(CalendarErr::CALENDAR_OR_CALENDAR_EVENT_NOT_FOUND);
+                }
+            }
+            throw ErrorHelper::getSherlError(
+                $err,
+                $this->errorFactory->create(CalendarErr::DELETE_CALENDAR_EVENT_FAILED)
+            );
+        }
+    }
+
+    /**
+     * Retrieves all calendar events based on the provided calendar ID and filters.
+     *
+     * @param string $calendarId The ID of the calendar to retrieve events from.
+     * @param GetCalendarEventForCalendarInputDto $filters The filters to apply when retrieving calendar events.
+     * @throws SherlException If there is an error retrieving calendar events.
+     * @return GetCalendarEventForCalendarResultsDto[] An array of GetCalendarEventForCalendarResultsDto objects representing the retrieved calendar events.
+     */
+    public function getAllCalendarEvents(string $calendarId, GetCalendarEventForCalendarInputDto $filters): array
```

Comment:
```suggestion
    public function getAllCalendarEvents(string $calendarId, GetCalendarEventForCalendarInputDto $filters): GetCalendarEventForCalendarResultsDto[]
```

Ne pas oublier de mettre l'import en haut du fichier s'il n'est pas déjà présent